src/robot/parsing/model/visitor.py

```diff
@@ -62,6 +84,6 @@ class ModelTransformer(ast.NodeTransformer, VisitorFinder):
     <https://docs.python.org/library/ast.html#ast.NodeTransformer>`__.
     """
 
-    def visit(self, node: Node):
-        visitor = self._find_visitor(type(node)) or self.generic_visit
-        return visitor(node)
+    def visit(self, node: Node) -> Node:
+        visitor = self._find_visitor(type(node)) or self.__class__.generic_visit
+        return visitor(self, node)
```

Comment:
This method could be probably implemented as `return VisitorFinder.visit(self, node)` to avoid duplication. Possibly it would also be enough to just change the order of inherited classes.